Feat/hng 46 faq accordion component

frank1003A opened the pull request:
<!-- Do not delete this PR template. Just edit it to include the required information -->

# Description

Closes #46

## What were you told to do?

Implement the FAQ Accordion component as described in [issue #46](https://github.com/hngprojects/hng_boilerplate_nextjs/issues/46).

## What did you do?

- Added the `FaqAccordion` component.
- Updated styles and classes for the accordion component.
- Added unit tests to verify accordion functionality.
- Ensured responsiveness of the accordion component.
- Updated the TypeScript interfaces and component props to support additional customizations.
- Fixed any issues related to UI interaction and state management.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

<!-- Mark all the applicable boxes. To mark the box as done follow the following conventions -->

<!--
[x] - Correct; marked as done
[X] - Correct; marked as done

[ ] - Not correct; marked as **not** done
-->

- [x ] My code follows the code style of this project.
- [ x] This PR does not contain plagiarized content.
- [x ] The title and description of the PR are clear and explain the approach.
- [ x] I am making a pull request against the **dev branch** (left side).
- [x ] My commit message style matches our requested structure.
- [ x] My code additions do not fail code linting checks or unit tests.
- [x ] I am only making changes to files I was requested to.

# Screenshots/Videos

![ts1](https://github.com/user-attachments/assets/580c1ae1-9b67-4986-a67c-f280aee84274)
![ts2](https://github.com/user-attachments/assets/9520f4ba-4150-4e9e-b2e6-6589893d150b)
![ts4](https://github.com/user-attachments/assets/fbce7598-c1dc-497f-89d1-d4384ca6b001)
![ts3](https://github.com/user-attachments/assets/c4e96ff3-cb51-4449-a915-daeeefc02b65)

frank1003A committed:
feat: Add FaqAccordion component with responsiveness and unit tests

- Created FaqAccordion component to display FAQ items using the Accordion component.
- Added responsive design to the FaqAccordion component, ensuring it adapts to different screen sizes with appropriate classes.
- Included optional className props for container, trigger, and content elements to allow custom styling.
- Updated FaqAccordion component to use clsx for conditional className handling.
- Added role="region" to the FaqAccordion container div for better accessibility and testing.
- Implemented unit tests using vitest to:
  - Verify correct rendering of FAQ questions and content.
  - Test the collapse and expand functionality of Accordion items.
  - Ensure the component renders correctly for different screen sizes.

frank1003A committed:
Merge remote-tracking branch 'upstream/dev' into feat/HNG-46-faq-accordion-component

frank1003A committed:
updates test file, run pnpm install

frank1003A committed:
chore: update responsive value, vitest snapshot update

frank1003A committed:
Merge remote-tracking branch 'upstream/dev' into feat/HNG-46-faq-accordion-component

frank1003A committed:
chore: delete dummy page

frank1003A committed:
fix: fix containerClassName props passing error

frank1003A committed:
chore: update lockfile

frank1003A committed:
Merge remote-tracking branch 'upstream/dev' into feat/HNG-46-faq-accordion-component

taiwonaf commented on src/components/common/FaqAccordion/FaqAcccordion.test.tsx:
```diff
@@ -0,0 +1,51 @@
+/* eslint-disable prettier/prettier */
```
why did you disable this ?

frank1003A replied:
Eslint configuration with local was giving issues. Fixed now

taiwonaf commented on src/components/common/FaqAccordion/FaqAcccordion.test.tsx:
```diff
@@ -0,0 +1,51 @@
+    expect.hasAssertions();
+    render(<FaqAccordion faqs={faqs} />);
+
+    // Check if the questions are rendered
```
remove comment

frank1003A replied:
done

taiwonaf commented on src/components/common/FaqAccordion/FaqAcccordion.test.tsx:
```diff
@@ -0,0 +1,51 @@
+      />,
+    );
+
+    // Check if the container class includes responsive width
```
remove

frank1003A replied:
done

frank1003A committed:
chore: update test file based on PR review

- Addressed feedback from PR review by https://github.com/taiwonaf
- Improved import statements and fixed sorting issues
- Refactored test cases for better readability and maintainability
- Ensured consistency with project coding standards

frank1003A committed:
Merge remote-tracking branch 'upstream/dev' into feat/HNG-46-faq-accordion-component

frank1003A committed:
update src/components/common/Faqs/index.tsx

frank1003A committed:
chore: delete duplicate FaqAccordion.test.tsx
